### PR TITLE
[DO NOT MERGE] revm executor

### PR DESCRIPTION
**Pull request by lightsing**

### CODE REVIEW

File: eth-types/src/revm.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+impl From<&TransactionTrace> for TxEnv {
+    fn from(tx: &TransactionTrace) -> Self {
+        TxEnv {
+            caller: Address::from(tx.from.to_fixed_bytes()),
+            gas_limit: tx.gas,
+            gas_price: U256::from_be_bytes(tx.gas_price.to_be_bytes()),
+            transact_to: match tx.to {
+                Some(to) => TransactTo::Call(Address::from(to.to_fixed_bytes())),
+                None => TransactTo::Create(CreateScheme::Create), /* FIXME: is this correct?
```
Comment (lispc) [inline eth-types/src/revm.rs:41]: of course tx.to is null means create scheme

**Comment by lispc** [comment]
generally good as the first POC.

( after reading this PR, i realized something interesting.   I am considering use standard ETH rpc for proving instead of our homemade "scroll-trace" format in future.  Yeah pre-state-tracer can be very useful.  But a tricky thing is the bytecodes.  Now we use StructLog.codes for this purpose.  But if we  switch to standard ETH rpc, we may need prestate trace + call trace? And collect contract addresses from call traces. Or  we can take a look at designs like this https://github.com/ethereum/go-ethereum/pull/28776 and https://github.com/ethereum/go-ethereum/pull/28640/files#diff-551c7db294c6206e4b3bd0c9553f33b3c931f6d64b5e16ac6531d016fb49794e ) 

zeth uses a "rpc-backed-remote-db" design to fill needed witness and bytecodes.

https://github.com/a16z/helios can also be useful

File: bus-mapping/src/precompile.rs [hunk 2]
```diff
@@ -28,46 +28,48 @@ pub(crate) fn execute_precompiled(
...
-        },
-    };
+            Err(err) => match err {
+                PrecompileError::OutOfGas => (vec![], gas, true, false),
+                _ => {
+                    log::warn!("unknown precompile err {err:?}");
+                    (vec![], gas, false, false)
+                }
+            },
+        };
```
Comment (lispc) [inline bus-mapping/src/precompile.rs:72]: why these changes are needed?

**Comment by lispc** [comment]
(keep this pr open, some codes are useful, we can merge them back into develop later)

File: eth-types/src/revm.rs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                .header
+                .mix_hash
+                .map(|h| B256::from(h.to_fixed_bytes())),
+            blob_excess_gas_and_price: None,
+        }
+    }
+}
+
+impl From<&TransactionTrace> for TxEnv {
+    fn from(tx: &TransactionTrace) -> Self {
```
Comment (lispc) [inline eth-types/src/revm.rs:34]: i see.. so here is the bridge/boudary between old ethers-rs and the new alloy-rs.  Not very complex.

Comment (lightsing) [reply eth-types/src/revm.rs:34]: It's not exist for alloy-rs, it's part of revm even in old version

File: bus-mapping/src/precompile.rs (hunk 2 above)
Comment (lightsing) [reply bus-mapping/src/precompile.rs:72]: the signature changed, Bytes version are different

Comment (lightsing) [reply bus-mapping/src/precompile.rs:72]: The signature changed, args and ret are Bytes instead of vec

**Comment by lispc** [comment]
on the future of this PR: we will keep both this PR and stateless-block-verifier, they both use scroll-revm.  The former can be combined with our testool driver to use the 18k tests to test l2geth <-> scroll-revm. The latter we will use it only for trace verification, no testool/golang codes.

**Pull request closed without merging**